packages/mui-base/src/useNumberInput/useNumberInput.types.ts

```diff
@@ -0,0 +1,193 @@
+import * as React from 'react';
+import { FormControlUnstyledState } from '../FormControlUnstyled';
+
+export type UseNumberInputChangeHandler = (
+  e: React.KeyboardEvent<HTMLInputElement>,
+  value: number | null,
+) => void;
+
+export interface UseNumberInputParameters {
+  /**
+   * The minimum value.
+   */
+  min?: number;
+  /**
+   * The maximum value.
+   */
+  max?: number;
+  /**
+   * The amount that the value changes on each increment or decrement.
+   */
+  step?: number;
+  /**
+   * Multiplier applied to `step` if the shift key is held while incrementing
+   * or decrementing the value. Defaults to `10`.
+   */
+  shiftMultiplier?: number;
+  /**
+   * The default value. Use when the component is not controlled.
+   */
+  defaultValue?: unknown;
+  /**
+   * If `true`, the component is disabled.
+   * The prop defaults to the value (`false`) inherited from the parent FormControl component.
+   */
+  disabled?: boolean;
+  /**
+   * If `true`, the `input` will indicate an error by setting the `aria-invalid` attribute.
+   * The prop defaults to the value (`false`) inherited from the parent FormControl component.
+   */
+  error?: boolean;
+  onBlur?: React.FocusEventHandler<HTMLInputElement>;
+  onClick?: React.MouseEventHandler;
+  onChange?: React.ChangeEventHandler<HTMLInputElement>;
+  onFocus?: React.FocusEventHandler;
+  /**
+   * Callback fired after the value is clamped and changes.
+   * Called with `undefined` when the value is unset.
+   */
+  onValueChange?: (value: number | undefined) => void;
```

Comment:
It's not immediately clear what's the difference between `onValueChange` and `onChange` (especially that onChange doesn't have any description)